tests/types/npm-test.json

```diff
@@ -254,6 +254,316 @@
+        "qualifiers": null,
+        "subpath": null
+      },
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Positive Test: A scoped npm purl with no version is valid",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/%40angular/core",
+      "expected_output": {
+        "type": "npm",
+        "namespace": "@angular",
+        "name": "core",
+        "version": null,
+        "qualifiers": null,
+        "subpath": null
+      },
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Negative Test: An npm purl with an empty version should fail",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/foobar@",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse due to an empty version string"
+    },
+    {
+      "description": "Negative Test: An npm purl with a name starting with a dot should fail",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/.foobar@1.0.0",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the name starts with a dot"
+    },
+    {
+      "description": "Negative Test: An npm purl with a name starting with an underscore should fail",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/_foobar@1.0.0",
+      "expected_output": null,
+      "expected_failure": true,
+      "expected_failure_reason": "Should fail to parse because the name starts with an underscore"
+    },
+    {
+      "description": "Edge Case Test: Version with pre-release and build metadata should be preserved",
+      "test_group": "advanced",
+      "test_type": "roundtrip",
+      "input": "pkg:npm/foobar@1.2.3-alpha.1+build.456",
+      "expected_output": "pkg:npm/foobar@1.2.3-alpha.1+build.456",
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Positive Test: Package name with dots and hyphens is valid",
+      "test_group": "advanced",
+      "test_type": "parse",
+      "input": "pkg:npm/my-package.js@1.0.0",
+      "expected_output": {
+        "type": "npm",
+        "namespace": null,
+        "name": "my-package.js",
+        "version": "1.0.0",
+        "qualifiers": null,
+        "subpath": null
+      },
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Normalization Test: Both scope and name with uppercase letters are lowercased",
+      "test_group": "advanced",
+      "test_type": "roundtrip",
+      "input": "pkg:npm/%40MyScope/MyPackage@1.0.0",
+      "expected_output": "pkg:npm/%40myscope/mypackage@1.0.0",
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
```

Comment:
This test makes the previous (incorrect) test about lowercasing the name redundant.